### PR TITLE
solucion al bug de guardias (no se pueden crear dos guardias para el mismo dia, no se puede crear una guardia para un dia pasado y si se puede crear una guardia para el mismo dia)

### PR DESCRIPTION
**Pull request by pablotocalini**

### CODE REVIEW

File: core/test/model/agenda.test.js [hunk 1]
```diff
@@ -112,6 +112,40 @@ describe("Agenda Test", () => {
...
+      //act
+      const canCreateGuard = agenda.canCreateGuard(TOMORROW);
+
+      //assert
+      expect(canCreateGuard).toBeFalsy();
+    });
+
+    test("can`t create a new guard for a past day", () => {
+      //act
+      const pastDay = agenda.isPastDate(YESTERDAY);
```
Comment (crudobaker) [inline core/test/model/agenda.test.js:142]: In this test You are not trying to create a guard. You should call agenda.createGuard()

File: core/test/model/agenda.test.js [hunk 2]
```diff
@@ -112,6 +112,40 @@ describe("Agenda Test", () => {
...
+      const guard2 = agenda.createGuard(TOMORROW);
+
+      //assert
+      expect(guard2).toThrow(
+        new Error("La guardia para esa fecha ya esta creada")
+      );
+    });
+    */
+
+    test("can`t create a new guard for a day with an already created guard2", () => {
```
Comment (crudobaker) [inline core/test/model/agenda.test.js:129]: Replace Guard2 for just guard

File: core/test/model/agenda.test.js [hunk 3]
```diff
@@ -112,6 +112,40 @@ describe("Agenda Test", () => {
       expect(agenda.findGuardById(guard.id)).toStrictEqual(guard);
     });
 
+    /* test("can`t create a new guard for a day with an already created guard", () => {
```
Comment (crudobaker) [inline core/test/model/agenda.test.js:115]: Lets try do not commit commented Code. Commented Code is dead Code.

File: core/src/date-utils.js [hunk 4]
```diff
@@ -38,3 +38,18 @@ export function toDate(date) {
...
   return new Date(Date.parse(finalDate));
 }
+
+export function isSameDate(date1, date2) {
+  return formatDate(date1) === formatDate(date2);
+}
+
+export function isFutureDate(date) {
+  const today = new Date();
+  if(today.getDate() == date.getDate()) {
```
Comment (crudobaker) [inline core/src/date-utils.js:48]: Why do we return true if the dates are equals ?🤔

File: core/src/data-source/repository.js [hunk 5]
```diff
@@ -59,4 +60,26 @@ export default class Repository {
   saveGuard(guard) {
     this.guards.push(guard);
   }
+
+  guardAlreadyCreated(date) {
+    //busco si hay una guardia creada igual a la que me pasan
```
Comment (crudobaker) [inline core/src/data-source/repository.js:65]: Remove commented code

File: bot/src/telegram/actions/create-guard.js [hunk 6]
```diff
@@ -17,7 +17,17 @@ const initCreateGuardFlow = (ctx) => {
   try {
     const { calendar } = ctx.state;
     calendar.onDateSelect((context, date) => {
-      askForConfirmation(context, date);
+      if(agenda.isPastDate(date)) {
```
Comment (crudobaker) [inline bot/src/telegram/actions/create-guard.js:20]: Lets pair about it!

File: core/test/model/agenda.test.js (hunk 1 above)
Comment (pablotocalini) [reply core/test/model/agenda.test.js:142]: ok, i will check it now

File: core/test/model/agenda.test.js (hunk 2 above)
Comment (pablotocalini) [reply core/test/model/agenda.test.js:129]: ok

Comment (pablotocalini) [reply core/test/model/agenda.test.js:129]: resolved

**Comment by crudobaker** [comment]
We close this PR cause we choose the solution thought the pairing session.

**Pull request closed without merging**